src/mastercard-auth.js

```diff
@@ -16,8 +16,23 @@ module.exports = function (context) {
 
   const mastercard = context.request.getEnvironmentVariable('mastercard');
 
-  if (mastercard && commaDecodedUrl.includes('mastercard.com')) {
+  if(mastercard){
     try {
+      if(mastercard.appliesTo){
+        var validConfiguration = false
+        mastercard.appliesTo.forEach((it) => {
+          if(commaDecodedUrl.includes(it)){
+            validConfiguration = true
+            return 
+          }
+        })
+        if(!validConfiguration){
+          return
+        }
+      }else{
+        throw Error("Please add at least one domain to appliesTo in the Mastercard environment settings")
```

Comment:
To avoid existing customers seeing an error after having updated the plugin (but also to have the same behavior as with the previous version), we could do something like this:

```javascript
// The plugin is only active for certain URIs. 
// By default, it only intercepts API calls sent to "mastercard.com" URIs
let appliesTo = mastercard.appliesTo ? mastercard.appliesTo : [ "mastercard.com" ]
```